<a href="https://colab.research.google.com/github/gilesitorr/Estadistica_Avanzada/blob/main/Parcial_3/Tarea_3/EA_parcial3_tarea3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# __Estadística Avanzada__
# Parcial 3 - Tarea 3
## _Gil Estéfano Rodríguez Rivera_

Todas las bases de datos utilizadas así como el desarrollo se encuentra en _GitHub_ en https://github.com/gilesitorr/Estadistica_Avanzada/tree/main/Parcial_3/Tarea_3

__Nota:__ En este Notebook se usó el kernel de R, aunque la extensión sea ipynb.

## Librerías

In [1]:
install.packages("pspearman")

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)



In [2]:
# Antes de comenzar, se importan las librerías a usar
library(dplyr) # Para manipular dataframes
library(pspearman) # Para el test de Spearman


Attaching package: ‘dplyr’


The following objects are masked from ‘package:stats’:

    filter, lag


The following objects are masked from ‘package:base’:

    intersect, setdiff, setequal, union




## Biostatistics: A Foundation for Analysis in the Health Sciences
### Wayne W. Daniel & Chad L. Cross

### Ejemplo 13.6.3


__The purpose of a study by researchers at the Cleveland (Ohio) Clinic (A-6) was to determine if the use
of Flomax® reduced the urinary side effects commonly experienced by patients following brachytherapy (permanent radioactive seed implant) treatment for prostate cancer. The following table
shows the American Urological Association (AUA) symptom index scores for two groups of subjects
after 8 weeks of treatment. The higher the AUA index, the more severe the urinary obstruction and
irritation__

__May we conclude, on the basis of these data, that the median AUA index in the Flomax® group differs
significantly from the median AUA index of the placebo group? Let $\alpha=0.05$.__

In [ ]:
# Se define el url donde está guardado el archivo csv a usar
url = "https://github.com/gilesitorr/Estadistica_Avanzada/raw/main/Parcial_3/Tarea_3/EXR_C13_S06_03.csv"

# Se lee el archivo csv desde el url que se definió
df <- read.csv(url, header=TRUE)

In [ ]:
# Se realiza una prueba de Mann-Whitney
wilcox.test(AUA~GROUP, data=df, exact=FALSE, correct=FALSE)


	Wilcoxon rank sum test

data:  AUA by GROUP
W = 826.5, p-value = 0.7596
alternative hypothesis: true location shift is not equal to 0


Con base a la prueba de Mann-Whitey, a una significancia de $\alpha=0.05$, se infiere que no hay suficiente evidencia para afirmar que hay una diferencia estadísticamente significativa en las medianas del índice AUA entre el grupo de control y el que consume Flomax ($p=0.7596$).

### Ejemplo 13.7.3


__For a sample of apparently normal subjects who served as controls in an experiment, the following
systolic blood pressure readings were recorded at the beginning of the experiment:__

162 177 151 167

130 154 179 146

147 157 141 157

153 157 134 143

141 137 151 161

__Can one conclude on the basis of these data that the population of blood pressures from
which the sample was drawn is not normally distributed with $\mu=150$ and $\sigma=12$? Determine
the p value.__

In [ ]:
# Se define la muestra
data <- data.frame(x=c(162, 177, 151, 167,
          130, 154, 179, 146,
          147, 157, 141, 157,
          153, 157, 134, 143,
          141, 137, 151, 161))

# Los parámetros de la distribución teórica
mu <- 150
std <- 12

In [ ]:
# Se realiza una prueba de Kologorov-Smirnov
ks.test(data$x, "pnorm", mean=mu, sd=std)

Warning message in ks.test(data$x, "pnorm", mean = mu, sd = std):
“ties should not be present for the Kolmogorov-Smirnov test”



	One-sample Kolmogorov-Smirnov test

data:  data$x
D = 0.13321, p-value = 0.8699
alternative hypothesis: two-sided


De acuerdo con la prueba de Kolmogorov-Smirnov, a una significancia de $\alpha=0.05$, se concluye que no hay evidencia suficiente para afirmar que existen valores para los que la distribución teórica y la muestra difieren de manera estadísticamente significativa ($D=0.1321$, $p=8699$).

### Ejemplo 13.8.3


__A study of young children by Flexer et al. (A-9) published in the Hearing Journal examines the
effectiveness of an FM sound field when teaching phonics to children. In the study, children in a
classroom with no phonological or phonemic awareness training (control) were compared to a class
with phonological and phonemic awareness (PPA) and to a class that utilized phonological and
phonemic awareness training and the FM sound field (PPA/FM). A total of 53 students from three
separate preschool classrooms participated in this study. Students were given a measure of phonemic
awareness in preschool and then at the end of the first semester of kindergarten. The improvement
scores are listed in the following table as measured by the Yopp–Singer Test of Phonemic
Segmentation.__

__Test for a significant difference among the three groups. Let $\alpha=0.05$.__

In [6]:
# Se define el url donde está guardado el archivo csv a usar
url = "https://github.com/gilesitorr/Estadistica_Avanzada/raw/main/Parcial_3/Tarea_3/EXR_C13_S08_03.csv"

# Se lee el archivo csv desde el url que se definió
df <- read.csv(url, header=TRUE)

In [7]:
# Se realiza una prueba de Kruskal-Wallis
kruskal.test(IMP ~ GROUP, data=df)


	Kruskal-Wallis rank sum test

data:  IMP by GROUP
Kruskal-Wallis chi-squared = 18.214, df = 2, p-value = 0.0001109


De acuerdo a la prueba de Kruskal-Wallis, a una significancia de $0.05$, se concluye que hay suficiente evidencia para afirmar que la diferencia entre las distribuciones de la mejora entre los 3 grupos es estadísticamente significativa ($X^2=18.2140$, $p=0.0001$).

### Ejemplo 13.8.6


__Hepatic g-glutamyl transpeptidase (GGTP) activity was measured in 22 patients undergoing
percutaneous liver biopsy. The results were as follows:__

__Can we conclude from these sample data that the average population GGTP level differs among the
five diagnostic groups? Let $\alpha=0.05$ and find the p value.__

In [12]:
# Se define el url donde está guardado el archivo csv a usar
url = "https://github.com/gilesitorr/Estadistica_Avanzada/raw/main/Parcial_3/Tarea_3/EXR_C13_S08_06.csv"

# Se lee el archivo csv desde el url que se definió
df <- read.csv(url, header=TRUE)

In [13]:
# e realiza una prueba de Kruskall-Wallis
kruskal.test(LEVEL ~ DIAG, data=df)


	Kruskal-Wallis rank sum test

data:  LEVEL by DIAG
Kruskal-Wallis chi-squared = 17.17, df = 4, p-value = 0.001792


De acuerdo a la prueba de Kruskal-Wallis, a una significancia de $0.05$, se concluye que hay suficiente evidencia para afirmar que la diferencia entre las distribuciones de los niveles de GGTP entre los 5 grupos es estadísticamente significativa ($X^2=17.17$, $p=0.0018$).

### Ejemplo 13.9.3


__Ten subjects with exercise-induced asthma participated in an experiment to compare the
protective effect of a drug administered in four dose levels. Saline was used as a control. The
variable of interest was change in FEV1 after administration of the drug or saline. The results
were as follows:__

__Can one conclude on the basis of these data that different dose levels have different effects?
Let $\alpha=0.05$ and find the p value.__

In [14]:
# Se define el url donde está guardado el archivo csv a usar
url = "https://github.com/gilesitorr/Estadistica_Avanzada/raw/main/Parcial_3/Tarea_3/EXR_C13_S09_03.csv"

# Se lee el archivo csv desde el url que se definió
df <- read.csv(url, header=TRUE)

In [25]:
# Se hace una prueba de Friedman
friedman.test(cbind(df$SALINE, df$T2, df$T10, df$T20, df$T40))


	Friedman rank sum test

data:  cbind(df$SALINE, df$T2, df$T10, df$T20, df$T40)
Friedman chi-squared = 29.528, df = 4, p-value = 6.107e-06


De acuerdo a la prueba de Friedmans, a una significancia de $0.05$, se concluye que hay suficiente evidencia para afirmar que los distintos tratamientos producen distintos cambios en el FEV1 del paciente ($X^2=29.5280$, $p\sim6\times10^{-6}$).

### Ejemplo 13.10.3


__The purpose of a study by Nozawa et al. (A-10) was to evaluate the outcome of surgical repair of pars
interarticularis defect by segmental wire fixation in young adults with lumbar spondylolysis. The
authors cite literature indicating that segmental wire fixation has been successful in the treatment of
nonathletes with spondylolysis and point out that no information existed on the results of this type of
surgery in athletes. In a retrospective study of subjects having surgery between 1993 and 2000, the
authors found 20 subjects who had undergone the surgery. The following table shows the age (years)
at surgery and duration (months) of follow-up care for these subjects.__

__May we conclude, on the basis of these data, that in a population of similar subjects there is an
association between age and duration of follow-up? Let $\alpha=0.05$.__

In [26]:
# Se define el url donde está guardado el archivo csv a usar
url = "https://github.com/gilesitorr/Estadistica_Avanzada/raw/main/Parcial_3/Tarea_3/EXR_C13_S10_03.csv"

# Se lee el archivo csv desde el url que se definió
df <- read.csv(url, header=TRUE)

In [31]:
# Se realiza un test de correlación de Spearman
spearman.test(df$FOLLOWUP, df$AGE)

Warning message in spearman.test(df$FOLLOWUP, df$AGE):
“Cannot compute exact p-values with ties”



	Spearman's rank correlation rho

data:  df$FOLLOWUP and df$AGE
S = 1305.8, p-value = 0.9411
alternative hypothesis: true rho is not equal to 0
sample estimates:
       rho 
0.01823019 


De acuerdo a la prueba de correlación de Spearman, a una significancia de $0.05$, se concluye que no hay suficiente evidencia para afirmar que las variables de edad y seguimiento tengan correlación ($r_s=0.0182$, $p=0.9411$).

### Ejemplo 13.10.6


__Seventeen patients with a history of congestive heart failure participated in a study to assess the
effects of exercise on various bodily functions. During a period of exercise the following data were
collected on the percent change in plasma norepinephrine (Y) and the percent change in oxygen
consumption (X):__

__On the basis of these data can one conclude that there is an association between the two variables? Let
$\alpha=0.05$.__

In [32]:
# Se define el url donde está guardado el archivo csv a usar
url = "https://github.com/gilesitorr/Estadistica_Avanzada/raw/main/Parcial_3/Tarea_3/EXR_C13_S10_06.csv"

# Se lee el archivo csv desde el url que se definió
df <- read.csv(url, header=TRUE)

In [34]:
# Se realiza un test de correlación de Spearman
spearman.test(df$X, df$Y)

Warning message in spearman.test(df$X, df$Y):
“Cannot compute exact p-values with ties”



	Spearman's rank correlation rho

data:  df$X and df$Y
S = 246.95, p-value = 0.00252
alternative hypothesis: true rho is not equal to 0
sample estimates:
      rho 
0.6973605 


De acuerdo a la prueba de correlación de Spearman, a una significancia de $0.05$, se concluye que hay suficiente evidencia para afirmar que las variables X y Y tienen correlación ($r_s=0.6974$, $p=0.0025$).

### Ejemplo 13.11.2


__The following are the body weights (grams) and total surface area (cm2
) of nine laboratory
animals:__

Body weight (X): 660.2 706.0 924.0 936.0 992.1 888.9 999.4 890.3 841.2

Surface area (Y): 781.7 888.7 1038.1 1040.0 1120.0 1071.5 1134.5 965.3 925.0

__Compute the slope estimator and two intercept estimators.__

In [26]:
# Se definen las medidas de ambas muestras
X <- c(660.2, 706.0, 924.0, 936.0, 992.1, 888.9, 999.4, 890.3, 841.2)
Y <- c(781.7, 888.7, 1038.1, 1040.0, 1120.0, 1071.5, 1134.5, 965.3, 925.0)

# Se calcula el estimador de la pendiente
S <- c()
for(i in 1:(length(X)-1)){
  for(j in (i+1):length(X)){
    S <- cbind(S, (Y[j]-Y[i])/(X[j]-X[i]) )
  }
}
(beta.1 <- median(S))

[1] 1.009368

In [27]:
# Se calcula el primer estimador de la intersección
first.intercept <- c()
for(i in 1:length(X)){
  first.intercept <- cbind(first.intercept,  Y[i]-beta.1*X[i])
}
(beta.0.A <- median(first.intercept))

[1] 115.3152

In [28]:
# Se calcula el segundo estimador de la intersección
second.intercept <- c()
for(i in 1:length(X)){
  for(j in i:length(X)){
    second.intercept <- cbind(second.intercept,  (first.intercept[i]+first.intercept[j])/2)
  }
}
(beta.0.B <- median(second.intercept))

[1] 115.5907

Según el método de Theil, la pendiente de la regresión es $\beta_1 = 1.0094$. Por su parte, los estimadores de la intersección son $\beta_{0,A}=115.3152$ y $\beta_{0,B}=115.5907$.